### ISSUE TITLE
projectGenerator binary distributed with linux (and vs) is a MACH-O binary.

### ISSUE THREAD

**Issue by bakercp** [openframeworks/projectGenerator#209]
Thus, it won't run on Linux.

```
bakercp@fmachine:~/Programming/of/of_v20190324_linux64gcc6_release/projectGenerator-linux64/resources/app/app$ file projectGenerator 
projectGenerator: Mach-O 64-bit x86_64 executable, flags:<NOUNDEFS|DYLDLINK|TWOLEVEL|WEAK_DEFINES|BINDS_TO_WEAK|PIE|HAS_TLV_DESCRIPTORS>
```

The MACH-O binary is in these two at least  I haven't checked others.

https://openframeworks.cc/versions/v0.10.1/of_v0.10.1_linux64gcc6_release.tar.gz
https://openframeworks.cc/ci_server/versions/nightly/of_v20190324_linux64gcc6_nightly.tar.gz

They run on macOS if libmodex.dylib is present.

**Comment by bakercp** [comment]
Also, this 

:~/tmp/of_v20190324_vs2017_release/projectGenerator-vs/resources/app/app$ file projectGenerator
projectGenerator: Mach-O 64-bit executable x86_64

**Comment by arturoc** [comment]
there shouldn't be any binary at all in linux since it won't be compatible between versions anyway so we compile it manually when running the compilePG.sh script on install. will take a look anyway to see why there's a binary

**Comment by arturoc** [comment]
not sure if it's related but even after compiling the project generator biunary in linux, the gui version will throw an error when trying to create a new project that it doesn't find the projectGenerator binary

**Comment by ofTheo** [comment]
<img width="983" alt="image" src="https://user-images.githubusercontent.com/144000/69359582-b5bfb680-0c3d-11ea-81e5-758140f305dc.png">

So I see the macOS binary in the projectgenerator-vs folder ( far right ) but it is alongside the .exe so not sure if it is actually getting called or just needs to be removed from the app/ folder 

Will see if I can track down in this repo if there was a recent change causing this issue.

**Comment by ofTheo** [comment]
I wonder if the issue for linux is: 
https://github.com/openframeworks/projectGenerator/blob/85d0a821a908ef5401a42462171875cce4a520ca/frontend/index.js#L769 

But in our repo, this file is a macOS binary file. 
https://github.com/openframeworks/projectGenerator/blob/master/frontend/app/projectGenerator

Maybe the file isn't getting overwritten? 

@arturoc do you see this with the nightly builds? 
@bakercp or anyone with Windows does this Windows PG in the nightlies work fine?

**Comment by ofTheo** [comment]
This is def a bit out of my area of understanding but it looks like buildPG.sh uploads the linux compiled PG to projectGenerator_builds/projectGenerator_linux on the test server. 

Then we only update the PG on release  ( not nightly builds ). 
See the last result of the search here: https://github.com/openframeworks/openFrameworks/search?q=projectGenerator_linux&unscoped_q=projectGenerator_linux 

In the create_package.sh which is called by the nightly_build.sh 
this line https://github.com/openframeworks/openFrameworks/blob/e6e4ee1da88f507f36bad503cd329b31eaadccac/scripts/dev/create_package.sh#L439

builds the front end but doesn't compile the commandline executable. So the macOS executable in resources/app/app doesn't get overwritten. 

So for linux we're missing a step in the create_package.sh that is similar to this for VS:
https://github.com/openframeworks/openFrameworks/blob/e6e4ee1da88f507f36bad503cd329b31eaadccac/scripts/dev/create_package.sh#L439

That explains why the VS packages have both the macOS binaries and the VS binary, but linux just has the macOS binary. 

**So the solution ( if I understand this right ) is when working with the repo copy the command line PG to `resources/app/app/projectGenerator` overwriting the macOS version. And for the nightly builds, fix the `create_package.sh` script [at these lines](https://github.com/openframeworks/openFrameworks/blob/e6e4ee1da88f507f36bad503cd329b31eaadccac/scripts/dev/create_package.sh#L439) to do the same on the server.**

**Comment by ofTheo** [comment]
So @prisonerjohn confirmed that on Windows the PG works fine. So its mostly just an issue with Linux it looks like. ( though we could probably do with removing the macOS binary either way )

**Comment by arturoc** [comment]
the binary for linux is compiled on peoples machines, not on the servers, so the problem is surely on the frontend code. and yes this is happening on the nightly builds.

**Comment by ofTheo** [comment]
@arturoc thanks! 
It has been pretty hard to follow how it works on all the different platforms. That makes sense though. 

In the frontend there are two different references to the pgApp location. 

Here it looks like it is referencing the one in `app/` folder for linux. 
https://github.com/openframeworks/projectGenerator/blob/master/frontend/index.js#L681
 
But here it looks like its expecting the PG to be installed in usr/local/bin/ 
https://github.com/openframeworks/projectGenerator/blob/master/frontend/index.js#L770

for both should it be looking in `apps/projectGenerator/commandLine/bin` for linux/linux64?
where the [compilePG.sh script](https://github.com/openframeworks/openFrameworks/blob/master/scripts/linux/compilePG.sh) builds the PG to?

**Comment by bakercp** [comment]
This appears to still be an issue with the latest release on Linux.

**Comment by ofTheo** [comment]
I think it could be a simple as removing this file from the PG repo:
https://github.com/openframeworks/projectGenerator/blob/master/frontend/app/projectGenerator

**Comment by bakercp** [comment]
The hash on that file matches the one I have ... so definitely part of the problem.  Not sure why it's being placed here `of_v0.11.2_linux64gcc6_release/projectGenerator-linux64/resources/app/app/projectGenerator`. It would be useful to have a compiled CLI distrubtued -- maybe I'm just pointing at the wrong one with my scripts?

**Comment by ofTheo** [comment]
@bakercp  

Do you compile the PG with and then select Y to install? 
https://github.com/openframeworks/openFrameworks/blob/master/scripts/linux/compilePG.sh#L17

Right now the [linux setup instructions](https://openframeworks.cc/setup/linux-install/) include that step as it seems it isn't as easy to distribute the linux executable. 

I think if you run that script it should pickup the right path to run the CLI from.

**Comment by arturoc** [comment]
Yes linux binaries won't work across versions so it needs to be compiled by the users. In any case there shouldn't be a binary there if there's one.
